javascript/hash-table/hashTable.js

```diff
@@ -29,17 +30,37 @@ class HashTable {
 
   set(key, val) {
     let index = this.hash(key) % this.size;
-    if (this.keys[key]) {
-      //go to the index in the bucket and find the entry where key === key, and replace the value
-
-    }
     if (this.buckets[index]) {
+      //go to the index in the bucket and find the entry where key === key, and replace the value
       if (this.keys[key]) {
         this.update(index, key, val);
-      } else this.buckets[index].push({ key, val});
+      } else this.buckets[index].push({ key, val });
     } else this.buckets[index] = [{key, val}];
     this.keys[key] = true;
   }
+
+  get(key){
+    let val = this.hash(key) % this.size;
+    let bucket = this.buckets[val];
+    if (!bucket) {
+      return null;
+    }
+
+    for(let i = 0; i < bucket.length; i++) {
+      if(bucket[i].key === key) {
+        return bucket[i].val;
+      }
+    }
+    return null;
+  }
+
+  has(key){
+    return this.keys[key] !== undefined;
+  }
+
+  keys(){
+    return Object.keys(this.keys);
```

Comment:
Your code is slightly confused because you have two things named 'keys', the constructor this.'keys,' which is an object, and the other is a function called 'keys'.  Try renaming the function to 'key'. That way, you don't have to find all the this.keys and change them. If you do that in your test, change the keys() to key()